core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1136,60 +1176,73 @@ write_files:
             - list
             - watch
 
-  - path: /srv/kubernetes/rbac/cluster-role-bindings/kubelet-bootstrap.yaml
+  - path: /srv/kubernetes/rbac/cluster-role-bindings/node-bootstrap.yaml
     content: |
         kind: ClusterRoleBinding
         apiVersion: rbac.authorization.k8s.io/v1beta1
         metadata:
-          name: kubelet-bootstrap
+          name: kube-aws:node-bootstrap
         subjects:
           - kind: Group
             namespace: '*'
             name: system:kubelet-bootstrap
         roleRef:
           kind: ClusterRole
-          name: kubelet-bootstrap
+          name: system:node-bootstrapper
           apiGroup: rbac.authorization.k8s.io
 {{ end }}
 {{ end }}
 
-  - path: /etc/kubernetes/manifests/kube-proxy.yaml
+  - path: /srv/kubernetes/manifests/kube-proxy-ds.yaml
     content: |
-        apiVersion: v1
-        kind: Pod
+        apiVersion: extensions/v1beta1
+        kind: DaemonSet
         metadata:
-          name: kube-proxy
-          namespace: kube-system
           labels:
             k8s-app: kube-proxy
-          annotations:
-            rkt.alpha.kubernetes.io/stage1-name-override: coreos.com/rkt/stage1-fly
-
+          name: kube-proxy
+          namespace: kube-system
         spec:
-          hostNetwork: true
-          containers:
-          - name: kube-proxy
-            image: {{.HyperkubeImage.RepoWithTag}}
-            command:
-            - /hyperkube
-            - proxy
-            - --master=http://127.0.0.1:8080
-            securityContext:
-              privileged: true
-            volumeMounts:
-            - mountPath: /etc/ssl/certs
-              name: ssl-certs-host
-              readOnly: true
-            - mountPath: /var/run/dbus
-              name: dbus
-              readOnly: false
-          volumes:
-          - hostPath:
-              path: /usr/share/ca-certificates
-            name: ssl-certs-host
-          - hostPath:
-              path: /var/run/dbus
-            name: dbus
+          selector:
+            matchLabels:
+              k8s-app: kube-proxy
+          template:
+            metadata:
+              labels:
+                k8s-app: kube-proxy
+            spec:
+              containers:
+              - name: kube-proxy
+                image: {{.HyperkubeImage.RepoWithTag}}
+                imagePullPolicy: IfNotPresent
+                command:
+                - /usr/local/bin/kube-proxy
+                - --cluster-cidr={{.PodCIDR}}
+                - --kubeconfig=/etc/kubernetes/proxy-kubeconfig.yaml
+                securityContext:
+                  privileged: true
```

Comment:
it is a linux capability which gives control over network subsystem. Basically instead of running fully privileged container we can run it with:

```
securityContext:
      capabilities:
        add:
        - CAP_NET_ADMIN
```

and should be enough